File: spec/lua/test_default_endian_expr_inherited.lua
local luaunit = require("luaunit")

require("default_endian_expr_inherited")

TestDefaultEndianExprInherited = {}

function TestDefaultEndianExprInherited:test_default_endian_expr_inherited()
    local r = DefaultEndianExprInherited:from_file("src/endian_expr.bin")

    luaunit.assertEquals(r.docs[1].indicator, "II")
    luaunit.assertEquals(r.docs[1].main.insides.some_int, 0x42)
    luaunit.assertEquals(r.docs[1].main.insides.more.some_int1, 0x4200)
    luaunit.assertEquals(r.docs[1].main.insides.more.some_int2, 0x42)
    luaunit.assertEquals(r.docs[1].main.insides.more.some_inst, 0x42)

    luaunit.assertEquals(r.docs[2].indicator, "MM")
    luaunit.assertEquals(r.docs[2].main.insides.some_int, 0x42)
    luaunit.assertEquals(r.docs[2].main.insides.more.some_int1, 0x42)
    luaunit.assertEquals(r.docs[2].main.insides.more.some_int2, 0x4200)
    luaunit.assertEquals(r.docs[2].main.insides.more.some_inst, 0x42000000)

    luaunit.assertEquals(r.docs[3].indicator, "XX")
    luaunit.assertEquals(r.docs[3].main.insides.some_int, 0x42)
    luaunit.assertEquals(r.docs[3].main.insides.more.some_int1, 0x42)
    luaunit.assertEquals(r.docs[3].main.insides.more.some_int2, 0x4200)
    luaunit.assertEquals(r.docs[3].main.insides.more.some_inst, 0x42000000)
end

File: spec/python/test_default_endian_expr_inherited.py
import unittest

from default_endian_expr_inherited import DefaultEndianExprInherited

class TestDefaultEndianExprInherited(unittest.TestCase):
    def test_default_endian_expr_inherited(self):
        r = DefaultEndianExprInherited.from_file("src/endian_expr.bin")

        self.assertEqual(r.docs[0].indicator, b"II")
        self.assertEqual(r.docs[0].main.insides.some_int, 0x42)
        self.assertEqual(r.docs[0].main.insides.more.some_int1, 0x4200)
        self.assertEqual(r.docs[0].main.insides.more.some_int2, 0x42)
        self.assertEqual(r.docs[0].main.insides.more.some_inst, 0x42)

        self.assertEqual(r.docs[1].indicator, b"MM")
        self.assertEqual(r.docs[1].main.insides.some_int, 0x42)
        self.assertEqual(r.docs[1].main.insides.more.some_int1, 0x42)
        self.assertEqual(r.docs[1].main.insides.more.some_int2, 0x4200)
        self.assertEqual(r.docs[1].main.insides.more.some_inst, 0x42000000)

        self.assertEqual(r.docs[2].indicator, b"XX")
        self.assertEqual(r.docs[2].main.insides.some_int, 0x42)
        self.assertEqual(r.docs[2].main.insides.more.some_int1, 0x42)
        self.assertEqual(r.docs[2].main.insides.more.some_int2, 0x4200)
        self.assertEqual(r.docs[2].main.insides.more.some_inst, 0x42000000)
